### PR TITLE
replace circleci with github actions for deploy

### LINKED ISSUE #187: Switch to GitHub Actions for deployment
Right now deployment uses a repo-specific deploy key (see https://circleci.com/docs/2.0/gh-bb-integration/#deployment-keys-and-user-keys) and it's broken for at least some people that are not me. The SSH deploy key is two years old, so should be renewed anyway.

GitHub Actions is way easier to deal with for deployment, see https://getnikola.com/blog/automating-nikola-rebuilds-with-github-actions.html. Preview builds are separate anyway in Netlify, so we should just drop CircleCI completely.

### PR DESCRIPTION
**Pull request by mattip**
Fixes #187 

Since this destructively replaces the gh-pages branch and a mistake will ruin the website, please carefully monitor the site while/after merging this. Probably best done together with publishing a new blog post.

Also does not cache after `pip install`, maybe that should be left for a follow on PR to minimize chances of doing damage.

### CODE REVIEW

**Comment by ccordoba12** [comment]
Hey @mattip, to activate Github Actions in a repo you need to create a PR from an in-repo branch. It doesn't work from forks.

**Comment by mattip** [comment]
I don't have permissions to push to this repo

**Comment by Carreau** [comment]
> Since this destructively replaces the gh-pages branch and a mistake will ruin the website, please carefully monitor the site while/after merging this. Probably best done together with publishing a new blog post.

Do you mean that this replace the history ? Is there any reason to do that ? 
Or that it `rm -rf *` in that branch before copying ?

Current version as of this writing is aca8d3f and we can likely create a duplicate of the current gh-pages branch if you mean that it will erase history.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  nikola_build:
+    runs-on: ubuntu-latest
+    name: 'Build pages'
+    steps:
+    - name: Check out
+      uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v2
+    - name: Install dependencies
+      run: python -m pip install -r requirements.txt ghp-import`
```
Comment (Carreau) [inline .github/workflows/deploy.yml:19]:
```suggestion
      run: python -m pip install -r requirements.txt ghp-import
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Build
+      run: nikola build
+    - name: Deploy
+      if: github.ref == 'refs/heads/main'
```
Comment (Carreau) [inline .github/workflows/deploy.yml:23]:
```suggestion
      if: github.ref == 'refs/heads/master'
```

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      shell: bash
+      # -m : commit message
+      # -p : push the branch
+      # -f : force the push to the repo
+      # -r : use this git remote for the push
+      # -b : use this branch to commit/push to
+      # <directory> : should match the OUTPUT_DIR from conf.py, 'output' by default
+      run: ghp-import -m"Automatic push by ghp-import" -f -p -r origin -b gh-pages output
```
Comment (Carreau) [inline .github/workflows/deploy.yml:31]:
```
$ ghp-import
Traceback (most recent call last):
  File "...bin/ghp-import", line 5, in <module>
    from ghp_import import main
ImportError: cannot import name 'main' from 'ghp_import' (.../lib/python3.8/site-packages/ghp_import/__init__.py)
```

**Comment by Carreau** [comment]
ghp-import really does not work, and remove files that are critical for the website to work (cname and others). 
Closing.

**Pull request closed without merging**